src/core/search/compressed_sorted_set.cc

```diff
@@ -195,6 +199,33 @@ size_t CompressedSortedSet::ByteSize() const {
   return diffs_.size();
 }
 
+void CompressedSortedSet::Merge(CompressedSortedSet&& other) {
+  for (int v : other)
+    Insert(v);
+}
+
+std::pair<CompressedSortedSet, CompressedSortedSet> CompressedSortedSet::Split() && {
+  DCHECK_GT(Size(), 5u);
+
+  CompressedSortedSet second(diffs_.get_allocator().resource());
+
+  // Move iterator to middle position and save size of diffs tail
+  auto it = begin();
+  std::advance(it, size_ / 2);
+  size_t keep_bytes = it.last_read_.data() - diffs_.data();
+
+  // Copy second half into second set
+  for (; it != end(); ++it)
+    second.Insert(*it);
```

Comment:
I think we can implement it with better complexity

Reply:
Linear, because tail insert optimization (the iterator is sorted). Will add a comment

Also I might need to reserve the memory ahead